### PR TITLE
Remove outdated Observable check in qml.equal

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**

If the first operation in `qml.equal` is an adjoint of an Observable, it skips the "do they match" check because of an outdated clause that was meant to allow comparison of instances of the removed `Tensor` and `Hamiltonian` classes. This means that the second operation is not confirmed to also be an `Adjoint` before jumping to the function that assumes this check has already been performed, and leads to bugs like this:

```
>>> qml.equal(qml.adjoint(H(0)), qml.Y(0))
AttributeError: 'PauliY' object has no attribute 'base'
```

and

```
>>> qml.equal(qml.adjoint(X(0)), qml.exp(qml.X(0), 2))
True
```

This came up in a comparison of to operations in a decomposition when updating `assert_valid` (#7099).

**Description of the Change:**

Removed the exception for if `op1` is an `Observable`.

**Benefits:**

Fixes this edge-case bug for comparing operators where `op1` is the adjoint of an observable, and `op2` is not an `Adjoint`.

**Possible Drawbacks:**

If the tests and I are both missing some edge case where this is still relevant, I guess we might be breaking something.

[sc-86861]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -698,6 +698,10 @@
 * `qml.qchem.givens_decomposition` no longer raises a `RuntimeWarning` when the input is a zero matrix.
   [#7053)](https://github.com/PennyLaneAI/pennylane/pull/7053)
 
+* Comparing an adjoint of an `Observables` with another `Operation` using `qml.equal` no longer incorrectly
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:701]:
```suggestion
* Comparing an adjoint of an `Observable` with another `Operation` using `qml.equal` no longer incorrectly 
```

**Review by albi3ro — APPROVED** [review]

**Review by andrijapau — APPROVED** [review]
Thanks 😄

File: tests/ops/functions/test_equal.py [hunk 2]
```diff
@@ -1832,6 +1833,18 @@ def test_adjoint_base_op_comparison_with_trainability(self):
...
         assert not qml.equal(op1, op2, check_interface=False, check_trainability=True)
 
+    def test_adjoint_observable_with_non_adjoint(self):
+        """Test that comparing an adjoint observable with a non-adjoint operation returns False"""
+
+        base = qml.X(0)
+        adj_op = qml.adjoint(base)
+
+        # if we get rid of Observable or stop having adjoint(Observable) be an Observable,
+        # this test is no longer relevant
```
Comment (JerryChen97) [inline tests/ops/functions/test_equal.py:1843]:
```suggestion
        # this test is no longer relevant. Ref: https://github.com/PennyLaneAI/pennylane/pull/7107/
```

**Review by JerryChen97 — APPROVED** [review]
Thanks a lot for cleaning this up! I don't have any blocking suggestions, just thinking of having a hard link in the comment back to this PR.
